src/BlockTypes/MiniCart.php

```diff
@@ -338,4 +403,53 @@ protected function get_markup( $attributes ) {
 			</div>
 		</div>';
 	}
+
+	/**
+	 * Return an instace of the CartController class.
+	 *
+	 * @return CartController CartController class instance.
+	 */
+	protected function get_cart_controller() {
+		return new CartController();
+	}
+
+	/**
+	 * Get array with data for handle the tax label.
+	 * the entire logic of this function is was taken from:
+	 * https://github.com/woocommerce/woocommerce/blob/e730f7463c25b50258e97bf56e31e9d7d3bc7ae7/includes/class-wc-cart.php#L1582
+	 *
+	 * @return array;
+	 */
+	protected function get_tax_label() {
+		$cart = WC()->cart;
+
+		if ( $cart->display_prices_including_tax() ) {
+			if ( $cart->get_subtotal_tax() > 0 && ! wc_prices_include_tax() ) {
```

Comment:
One case that is not working for me is that, if you start with an empty cart, now it correctly doesn't include the `(incl. VAT)` string, but when you add a product to your Cart, the string doesn't appear when it should. Instead, you need to refresh the page.

Reply:
Oh, great catch thanks!

The problem was the check `$cart->get_subtotal_tax() > 0`.

The BE always has to return the information for handling the tax label. The client display or not the label based on whether there are products in the shopping cart